src/main/java/decodes/decoder/MetarProcess.java

```diff
@@ -0,0 +1,729 @@
+	private Calendar cal = null;
+	private char unitsFamily = 'E';
+	private RawMessage rawmsg = null;
+	private String tzID = "UTC";
+	private int qualityFlags = 0;
+	private String duration = null;
+	private int sensorNumber = -1;
+	private Platform platform = null;
+	private IntervalIncrement intinc = null;
+	private DecodedMessage decmsg = null;
+
+
+	// Where we are in the line. Used to ignore comments and bad lines
+	enum LineContext
+	{
+		StartOfLine, FormatType, FormatExt, IgnoreLine, InComment, Process
+	};
+
+	public MetarProcess()
+	{
+		super();
+	}
+	
+	public DecodesFunction makeCopy()
+	{
+		return new MetarProcess();
+	}
+
+	public void enableMatchShefPE() { matchShefPE = true; }
+	
+	public String getFuncName() { return module; }
+
+	/**
+	 * No arguments expected
+	 */
+	public void setArguments(String argString, DecodesScript script)
+	{
+	}
+	
+	public void execute(DataOperations dd, DecodedMessage msg)
+		throws DecoderException
+	{
+		this.decmsg = msg;
+		LineContext lineContext = LineContext.Process;
+
+		platform = msg.getPlatform();
+		if (platform == null)
+			throw new DecoderException(module + " function cannot be called with null platform.");
+		PlatformConfig config = platform.getConfig();
+		if (config == null)
+			throw new DecoderException(module + " function cannot be called with null config.");
+
+		rawmsg = msg.getRawMessage();
+
+		if (rawmsg.toString().length() <= 10)
+		{
+			throw new DecoderException("Cannot process METAR data without at least 10 characters ");
+		}
+		String msgString = rawmsg.toString();
+		String items[] = msgString.split(" ");
+		String icao="";
+		if( items.length > 0)
+		{
+			icao = items[0];
+		}
+
+		if( icao.length() != 4)
+		{
+			throw new DecoderException("Cannot find METAR station ID (4 characters long) '"+msgString+"'");
+		}
+
+		//MetarService service = MetarService.getInstance();
+		//Metar metar = service.retrieveFromAirport(icao);
+
+		Variable v = rawmsg.getPM(ShefPMParser.PM_MESSAGE_TYPE);
+		char formatType = v.getStringValue().charAt(0);
+		boolean isRevised = false;
+
+		// Set up calendar to process times as we step through message.
+		v = rawmsg.getPM(ShefPMParser.PM_TIMEZONE); // ShefPMParser sets this.
+		if (v != null)
+			tzID = v.toString();
+		
+		// Default is english units
+		unitsFamily = 'E';
+		duration = null;
+		sensorNumber = -1;
+
+		// All times will be completely assigned, so set timer to complete.
+		// Otherwise, after calling this method, the code will attempt to 'fixup'
+		// times.
+		debugSdf.setTimeZone(TimeZone.getTimeZone(tzID));
+		msg.getTimer().setTimeZoneName(tzID);
+		msg.getTimer().setComplete(msg.getMessageTime());
+		cal = msg.getTimer().getCalendar();
+
+		qualityFlags = shefQuality2Flags('Z'); // default is no qual-checks performed
+		
+		String msgData = new String(rawmsg.getData());
+		String shefdata = msgData.substring(rawmsg.getHeaderLength());
```

Comment:
shef? I assume you copied this from ShefParser and just haven't updated this method yet?

Reply:
Yes, just starting with a copy of ShefParser.